src/features/Instructors/ManageInstructors/ListInstructors.jsx

```diff
@@ -1,30 +1,97 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useLocation } from 'react-router-dom';
 import PropTypes from 'prop-types';
 
-import { Container, Spinner, Row } from '@edx/paragon';
-
-const ListInstructors = ({ instructors, isLoading }) => (
-  <Container size="xl" className="p-4 mt-3 instructors-content">
-    <Row className="justify-content-center my-4 my-3 px-3">
-      <h3 className="pb-2 col-12 px-0">Currently assigned instructors</h3>
-      {isLoading && (
-        <div className="w-100 h-100 d-flex justify-content-center align-items-center">
-          <Spinner
-            animation="border"
-            className="mie-3"
-            screenReaderText="loading"
-          />
-        </div>
-      )}
-      {instructors.length > 0 && (
-        <div className="list-instructors col-12 px-0">
-          {instructors.map(instructor => <p className="list-item" key={instructor}>{instructor}</p>)}
-        </div>
-      )}
-      {instructors.length === 0 && !isLoading && <p className="empty-list m-0 py-2 px-0 col-12">No records found.</p>}
-    </Row>
-  </Container>
-);
+import {
+  Container, Spinner, Row, useToggle,
+} from '@edx/paragon';
+import { logError } from '@edx/frontend-platform/logging';
+import { Button } from 'react-paragon-topaz';
+import DeleteModal from 'features/Common/DeleteModal';
+
+import { assignInstructors, fetchInstructorsOptionsData } from 'features/Instructors/data';
+import { initialPage } from 'features/constants';
+
+const ListInstructors = ({ instructors, isLoading }) => {
+  const location = useLocation();
+  const dispatch = useDispatch();
+  const institution = useSelector((state) => state.main.selectedInstitution);
+  const queryParams = new URLSearchParams(location.search);
+  const classId = queryParams.get('classId')?.replaceAll(' ', '+');
+  const [isOpenModal, openModal, closeModal] = useToggle(false);
+  const [instructorName, setInstructorName] = useState('');
+  const [instructorUsername, setInstructorUsername] = useState('');
+
+  const handleDeleteIcon = (instructorInfo) => {
+    setInstructorName(instructorInfo.instructorName);
+    setInstructorUsername(instructorInfo.instructorUsername);
+    openModal();
+  };
+
+  const handleDeleteInstructor = async () => {
+    try {
+      const instructorData = {
+        unique_student_identifier: instructorUsername,
+        rolename: 'staff',
+        action: 'revoke',
+        class_id: classId,
+      };
+
+      await dispatch(assignInstructors(instructorData));
+      dispatch(fetchInstructorsOptionsData(institution.id, initialPage, { limit: false, class_id: classId }));
+    } catch (error) {
+      logError(error);
```

Comment:
If the request fails, the user will not have any kind of feedback? I think the toast notification could be a good option, what do u think?